core/src/main/java/org/mskcc/cbio/oncokb/bo/impl/AlterationBoImpl.java

```diff
@@ -60,7 +60,8 @@ public Alteration findAlteration(Gene gene, AlterationType alterationType, Strin
         if (gene != null && consequence != null && !consequence.getTerm().equals("NA")) {
             if (alterations != null && alterations.size() > 0) {
                 for (Alteration alteration : alterations) {
-                    if (alteration.getGene().equals(gene) && alteration.getConsequence() != null && alteration.getConsequence().equals(consequence)) {
+                    if ((alteration.getGene().getEntrezGeneId() != null && alteration.getGene().getEntrezGeneId().equals(gene.getEntrezGeneId()) || alteration.getGene().getHugoSymbol() != null && alteration.getGene().getHugoSymbol().equals(gene.getHugoSymbol())) 
```

Comment:
Why changed the logic?

Reply:
The original logic was "alteration.getGene().equals(gene)" and it kept returning false. This is the problem I asked you earlier. As I understand, both gene and alteration came from the same cache,  I couldn't figure out why they are different. Sorry it is bad practice to change things I don't understand to just make it work. I will ask you more about this issue tomorrow.